cuqi/sampler/_gibbs.py

```diff
@@ -0,0 +1,224 @@
+from cuqi.distribution import JointDistribution
+from cuqi.sampler import Sampler
+from cuqi.samples import Samples
+from typing import Dict, Union
+import numpy as np
+import sys
+
+
+class Gibbs:
+    """
+    Gibbs sampler for sampling a joint distribution.
+
+    Gibbs sampling samples the variables of the distribution sequentially,
+    one variable at a time. The sampling of each variable is done by sampling
+    from the conditional distribution of that variable given the values of
+    the other variables. This is often a very efficient way of sampling
+    from a joint distribution if the conditional distributions are easy to
+    sample from.
```

Comment:
I was thinking this description: `Gibbs sampling samples the variables of the distribution sequentially,
    one variable at a time.` needs to clarify that what is meant by a variable is the entire possibly multi-variate variable x and not x components: x_i. (similar comment about the tutorial introduction)

Reply:
Added a comment on that. Thanks